runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -21,64 +24,77 @@
 @Slf4j
 public class ManagementView extends AbstractView {
 
-    private final long layoutRefreshTimeout = 500;
+    static final Duration WORKFLOW_TIMEOUT = Duration.ofMinutes(10);
+    static final Duration QUERY_INTERVAL = Duration.ofSeconds(1);
+    static final int WORKFLOW_RETRY = 3;
 
     public ManagementView(@NonNull CorfuRuntime runtime) {
         super(runtime);
     }
 
-    /**
-     * Checks if a given workflow with the given workflow ID is active or not.
-     * If returns True - the workflow is active
-     * If returns False - the workflow is either completed successfully or failed.
-     *
-     * @param server     Node endpoint on which the workflow is running.
-     * @param workflowId Workflow UUID
-     * @return True if active. Else False
-     */
-    public boolean isWorkflowActive(String server, UUID workflowId) {
-        return layoutHelper(l -> {
-            QueryResponse response = runtime.getRouter(server)
-                    .getClient(ManagementClient.class)
-                    .queryRequest(workflowId);
-            log.info("isWorkflowActive: {} on server:{} = {}",
-                workflowId, server, response.isActive());
-            return response.isActive();
-        });
+    void waitForWorkflow(UUID workflow, ManagementClient client) {
```

Comment:
You should mark this as private

Reply:
Also, shouldn't it return whether the workflow is finished or whether it failed to query?

Reply:
I'll rethrow the exception, that way if waitForWorkflow finishes then it finished waiting, otherwise if it throws an exception then it means that if failed to query. Based on that i'll retry.

Reply:
Done.